api/src/main/java/org/openmrs/Obs.java

```diff
@@ -106,55 +133,91 @@
 	 * 
 	 * @see #isObsGrouping() (??)
 	 */
+	@JoinColumn(name = "obs_group_id")
+	@ManyToOne
 	protected Obs obsGroup;
 	
 	/**
 	 * The list of obs grouped under this obs.
 	 */
 	@AllowDirectAccess
+	@JoinColumn(name = "obs_group_id")
+	@OneToMany(cascade = CascadeType.REMOVE)
+	@OrderBy("obs_id")
+	@BatchSize(size = 25)
 	protected Set<Obs> groupMembers;
 	
+	@JoinColumn(name = "value_coded")
+	@ManyToOne
 	protected Concept valueCoded;
 	
+	@JoinColumn(name = "value_coded_name_id")
+	@ManyToOne
 	protected ConceptName valueCodedName;
 	
+	@JoinColumn(name = "value_drug")
+	@ManyToOne
 	protected Drug valueDrug;
 	
+	@Column(name = "value_group_id")
 	protected Integer valueGroupId;
 	
+	@Column(name = "value_datetime", length = 19)
 	protected Date valueDatetime;
 	
+	@Column(name = "value_numeric", length = 22)
 	protected Double valueNumeric;
 	
+	@Column(name = "value_modifier", length = 2)
 	protected String valueModifier;
 	
+	@Column(name = "value_text", length = 65535)
 	protected String valueText;
 	
+	@Column(name = "value_complex")
 	protected String valueComplex;
 	
 	// ComplexData is not persisted in the database.
 	protected transient ComplexData complexData;
 	
+	@Column(name = "comments")
 	protected String comment;
 	
-	protected transient Integer personId;
```

Comment:
Having transient on this field as before was wrong because this is a persistent field.

Reply:
How was it able to persist values into the database when it was transient?

Reply:
Recall that we were using xml before, the field was mapped in the hbm file and that's why.

Reply:
This property duplicates the person property for for speed and simplicity reasons as you can see from the javadoc of the setter and getter methods. And this was done for serialisation purposes. That is why i feel like we should not remove it using this pull request.

Reply:
What am saying is that the field has always been persistent, meaning the transient keyword was redundant, you only add a transient keyword to none persistent fields. If you have an OpenMRS instance you will realize that it is written to the DB.

Reply:
See https://github.com/openmrs/openmrs-core/blob/1.11.x/api/src/main/resources/org/openmrs/api/db/hibernate/Obs.hbm.xml#L31

Reply:
I understand we added it for serialization purposes but persistence frameworks use the same keyword to determine which fields to persist vs not persist, if we don't remove it, the field is excluded from insert statements yet it is required, so you can't save an Obs to the DB.

If we don't want to serialize it, we would have to just update the serializer to explicitly exclude it.

Reply:
Does this mean that if we leave that transient keyword then we cannot save an Obs?

Reply:
Yes, we can't because just like a serializer, JPA frameworks also ignore the field in the insert statement yet it is not nullable in the DB. The reason why it was working before is because when using xml, hibernate builds the class metadata from the hbm file so it never discovers the transient keyword whereas with annotations it inspects the class' fields using reflection and ends up seeing the transient keyword.